UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Helpers/NMHelper.cs

```diff
@@ -41,6 +41,40 @@ public static List<ShowFilesResponseModel> ListFilesResponse(BatchSearchResponse
             return listshowFilesResponseModels;
         }
 
+        public static List<ShowFilesResponseModel> GetShowFilesResponseModel(List<BatchDetails> batchDetails)
+        {
+            List<ShowFilesResponseModel> listshowFilesResponseModels = new();
+            foreach (BatchDetails item in batchDetails)
+            {
+                foreach (BatchDetailsFiles file in item.Files)
+                {
+                    item.Attributes.Add(new BatchDetailsAttributes { Key = "BatchPublishedDate", Value = item.BatchPublishedDate.ToString() });
+                    listshowFilesResponseModels.Add(new ShowFilesResponseModel
+                    {
+                        Attributes = item.Attributes,
+                        BatchId = item.BatchId,
+                        Filename = file.Filename,
+                        FileDescription = Path.GetFileNameWithoutExtension(file.Filename),
+                        FileExtension = Path.GetExtension(file.Filename),
+                        FileSize = file.FileSize,
+                        FileSizeinKB = FileHelper.FormatSize((long)file.FileSize),
+                        MimeType = file.MimeType,
+                        Links = file.Links
+                    });
+                }
+            }
+            return listshowFilesResponseModels;
+        }
+
+        public static List<ShowFilesResponseModel> ListFilesResponseLeisure(BatchSearchResponse searchResult)
+        {
+
+            List<ShowFilesResponseModel> listshowFilesResponseModels  = GetShowFilesResponseModel(searchResult.Entries);
+            
+            return listshowFilesResponseModels.OrderByDescending(x => Convert.ToDateTime(x.Attributes.FirstOrDefault(y => y.Key == "BatchPublishedDate").Value)).GroupBy(x => x.Attributes.FirstOrDefault(y => y.Key == "Chart").Value)
+            .Select(grp => grp.First()).OrderBy(x=>x.FileDescription).ToList();
```

Comment:
If these attributes should never be null then it should be:
```suggestion
            return listshowFilesResponseModels.OrderByDescending(x => Convert.ToDateTime(x.Attributes.First(y => y.Key == "BatchPublishedDate").Value)).GroupBy(x => x.Attributes.First(y => y.Key == "Chart").Value)
            .Select(grp => grp.First()).OrderBy(x=>x.FileDescription).ToList();
```

Reply:
Kept is as FirstOrDefault and added null check to keep it in sync with rest of the code.